lib/read.js

```diff
@@ -346,30 +348,53 @@ var Read = Juttle.proc.source.extend({
         if (this.tailOptimized) {
             points.reverse();
         }
+        this.convertDatesToMoments(points);
+
+        // optimized reduce without reduce_every requires no timestamp
+        if (this.timeField &&
+            (this.optimization_info.type !== 'reduce' || this.optimization_info.reduce_every)
+        ) {
+            points = this.parseTime(points, this.timeField);
+        }
+
+        return points;
+    }
 
-        return this.parseTime(points, this.timeField);
-    },
+    convertDatesToMoments(points) {
```

Comment:
I'd like to stick to the current simple traversal until there is reason to support nested dates?